<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />




<title>Perform ordination</title>

<script src="site_libs/header-attrs-2.16/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/bootstrap.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/default.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DietR Package</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    ASA24
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="ASA24_1.html">Load and clean ASA24 data</a>
    </li>
    <li>
      <a href="ASA24_2.html">Calculate totals by occasion</a>
    </li>
    <li>
      <a href="ASA24_3.html">Data overview</a>
    </li>
    <li>
      <a href="ASA24_4.html">Percent kcal of CARB, PROT, TFAT</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="ASA24_5.html">Prepare for clustering</a>
    </li>
    <li>
      <a href="ASA24_6_PCA.html">PCA</a>
    </li>
    <li>
      <a href="ASA24_7_PCA_results.html">PCA results</a>
    </li>
    <li>
      <a href="ASA24_8_k-means.html">K-means</a>
    </li>
    <li class="divider"></li>
    <li>
      <a href="ASA24_9_create_food_tree.html">Create food tree</a>
    </li>
    <li>
      <a href="ASA24_10_viz_food_tree.html">Visualize food tree</a>
    </li>
    <li>
      <a href="ASA24_11_ordination.html">Ordination</a>
    </li>
    <li>
      <a href="ASA24_12_viz_ordination.html">Visualize ordination</a>
    </li>
  </ul>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    NHANES
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="NHANES_1_load_and_clean.html">Load and clean NHANES data - 1</a>
    </li>
    <li>
      <a href="NHANES_2_load_and_clean_2.html">Load and clean NHANES data - 2</a>
    </li>
  </ul>
</li>
<li>
  <a href="Links.html">Links</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Perform ordination</h1>

</div>


<p>In this section, we will take the phylogeny of food items into account in clustering individuals according to their dietary data. In order to do so, we will use <a href="https://joey711.github.io/phyloseq/index.html">the phyloseq package</a>, which uses phylogeny of microbes and their abundance. We will replace microbes with food items consumed by our dietary study participants.</p>
<p>Name the path to DietR directory where input files are pulled.</p>
<pre class="r"><code>  main_wd &lt;- &quot;~/GitHub/DietR&quot;</code></pre>
<p><br></p>
<p>If you have not downloaded and installed the phyloseq package yet, you can do so by first installing BiocManager (if you have not done so):</p>
<pre class="r"><code>  if (!require(&quot;BiocManager&quot;, quietly = TRUE)) install.packages(&quot;BiocManager&quot;)</code></pre>
<p>Then download and install the phyloseq package.</p>
<pre class="r"><code>  BiocManager::install(&quot;phyloseq&quot;)</code></pre>
<p><br></p>
<p>Load the necessary packages.</p>
<pre class="r"><code>  library(phyloseq)
  library(ggtree)</code></pre>
<p>Load the necessary functions.</p>
<pre class="r"><code>  source(&quot;lib/specify_data_dir.R&quot;)
  source(&quot;lib/ordination.R&quot;)</code></pre>
<p>You can come back to the main directory by:</p>
<pre class="r"><code>  setwd(main_wd)</code></pre>
<p><br></p>
<div id="create-a-phyloseq-object-for-ordination" class="section level2">
<h2>Create a phyloseq object for ordination</h2>
<div id="load-the-necessary-files-for-creating-a-phyloseq-object" class="section level3">
<h3>Load the necessary files for creating a phyloseq object</h3>
<p>Specify the directory where the data is.</p>
<pre class="r"><code>SpecifyDataDirectory(directory.name= &quot;eg_data/VVKAJ/&quot;)</code></pre>
<!-- [NOTE] SpecifyDataDirectory function is not executed by R markdown, but just showing  -->
<!--  the code so that this tutorial will have the same code as the R script.  -->
<div id="food" class="section level4">
<h4>Food</h4>
<p>Load food OTU table - this is our food OTU data</p>
<pre class="r"><code>food &lt;- read.delim(&quot;Foodtree/VVKAJ_Items_f_id_s_m_ff_reduced_4Lv.food.otu.txt&quot;, row.names = 1)</code></pre>
<p>“food” is a matrix of Food descriptions (rows) x SampleID (columns) with food taxonomy info at the end.</p>
<pre class="r"><code>head(food, 3)</code></pre>
<details>
<summary>
Click to expand output
</summary>
<pre><code>##                       vvkaj.00037 vvkaj.00002 vvkaj.00023 vvkaj.00029
## Milk NFS                      244           0           0         0.0
## Milk cows fluid whole           0         183         488       195.2
## Milk cows fluid 2 fat         244           0           0         0.0
##                       vvkaj.00044 vvkaj.00045 vvkaj.00022 vvkaj.00024
## Milk NFS                      0.0           0           0           0
## Milk cows fluid whole       170.8         244           0           0
## Milk cows fluid 2 fat         0.0           0         488         366
##                       vvkaj.00003 vvkaj.00016 vvkaj.00032 vvkaj.00038
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00039 vvkaj.00007 vvkaj.00034 vvkaj.00004
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00005 vvkaj.00011 vvkaj.00030 vvkaj.00010
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00018 vvkaj.00033 vvkaj.00009 vvkaj.00026
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00008 vvkaj.00031 vvkaj.00012 vvkaj.00027
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00042 vvkaj.00025 vvkaj.00041 vvkaj.00006
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00015 vvkaj.00013 vvkaj.00043 vvkaj.00028
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00040 vvkaj.00014 vvkaj.00001 vvkaj.00021
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00036 vvkaj.00019 vvkaj.00020 vvkaj.00035
## Milk NFS                        0           0           0           0
## Milk cows fluid whole           0           0           0           0
## Milk cows fluid 2 fat           0           0           0           0
##                       vvkaj.00017
## Milk NFS                        0
## Milk cows fluid whole           0
## Milk cows fluid 2 fat           0
##                                                                                                                                       taxonomy
## Milk NFS                                                         L1_Milk_and_Milk_Products;L2_Milks_and_milk_drinks;L3_Milk_fluid;L4_;Milk_NFS
## Milk cows fluid whole L1_Milk_and_Milk_Products;L2_Milks_and_milk_drinks;L3_Milk_fluid;L4_Milk_fluid_cow_goat_buttermilk;Milk_cows_fluid_whole
## Milk cows fluid 2 fat L1_Milk_and_Milk_Products;L2_Milks_and_milk_drinks;L3_Milk_fluid;L4_Milk_fluid_cow_goat_buttermilk;Milk_cows_fluid_2_fat</code></pre>
</details>
<p><br />
Format the food file and create an otu_table called ‘OTU’.</p>
<pre class="r"><code>PrepFood(data= food)</code></pre>
<p><br></p>
</div>
<div id="taxonomy-tax" class="section level4">
<h4>Taxonomy (tax)</h4>
<pre class="r"><code>tax &lt;- read.delim(&quot;Foodtree/VVKAJ_Items_f_id_s_m_ff_reduced_4Lv.tax.txt&quot;)</code></pre>
<p>Format the tax file and create a taxonomy table called ‘TAX’.</p>
<pre class="r"><code>PrepTax(data= tax)</code></pre>
<p><br></p>
</div>
<div id="sample-users-metadata" class="section level4">
<h4>Sample (Users’ metadata)</h4>
<pre class="r"><code>meta &lt;- read.table( &quot;ind_metadata_UserxDay.txt&quot;, sep=&quot;\t&quot;, header=T)</code></pre>
<p>Format the metadata file and save it as ‘SAMPLES’.</p>
<pre class="r"><code>PrepMeta(data= meta)</code></pre>
<p><br></p>
</div>
<div id="food-tree" class="section level4">
<h4>Food tree</h4>
<pre class="r"><code>foodtree &lt;- read_tree(&quot;Foodtree/VVKAJ_Items_f_id_s_m_ff_reduced_4Lv.tree.nwk&quot;)</code></pre>
<p>It is OK to see a message saying that<br />
“Found more than one class”phylo” in cache; using the first, from namespace ‘phyloseq’<br />
Also defined by ‘tidytree’“.</p>
<p>Format food tree and save it as ‘TREE’.</p>
<pre class="r"><code>PrepTree(data=foodtree)</code></pre>
<p>It is OK to see the same message as the previous line.</p>
<table style="width:6%;">
<colgroup>
<col width="5%" />
</colgroup>
<tbody>
<tr class="odd">
<td>#### Make a phyloseq object with OTU, TAX, SAMPLES, and TREE loaded above It is OK to see the same message as the previous line.</td>
</tr>
<tr class="even">
<td><code>r phyfoods &lt;- phyloseq(OTU, TAX, SAMPLES, TREE)</code></td>
</tr>
<tr class="odd">
<td>Check your metadata - show the sample names and ensure they are vvkaj.00xxx.</td>
</tr>
<tr class="even">
<td><code>r sample_names(phyfoods)</code></td>
</tr>
<tr class="odd">
<td><code>##  [1] "vvkaj.00037" "vvkaj.00002" "vvkaj.00023" "vvkaj.00029" "vvkaj.00044" ##  [6] "vvkaj.00045" "vvkaj.00022" "vvkaj.00024" "vvkaj.00003" "vvkaj.00016" ## [11] "vvkaj.00032" "vvkaj.00038" "vvkaj.00039" "vvkaj.00007" "vvkaj.00034" ## [16] "vvkaj.00004" "vvkaj.00005" "vvkaj.00011" "vvkaj.00030" "vvkaj.00010" ## [21] "vvkaj.00018" "vvkaj.00033" "vvkaj.00009" "vvkaj.00026" "vvkaj.00008" ## [26] "vvkaj.00031" "vvkaj.00012" "vvkaj.00027" "vvkaj.00042" "vvkaj.00025" ## [31] "vvkaj.00041" "vvkaj.00006" "vvkaj.00015" "vvkaj.00013" "vvkaj.00043" ## [36] "vvkaj.00028" "vvkaj.00040" "vvkaj.00014" "vvkaj.00001" "vvkaj.00021" ## [41] "vvkaj.00036" "vvkaj.00019" "vvkaj.00020" "vvkaj.00035" "vvkaj.00017"</code></td>
</tr>
<tr class="even">
<td>Show metadata.</td>
</tr>
<tr class="odd">
<td><code>r head(sample_data(phyfoods), n=3)</code></td>
</tr>
<tr class="even">
<td><code>##                SampleID UserName StudyDayNo       Diet Gender Age Weight Height ## vvkaj.00037 vvkaj.00037 VVKAJ113          1       Keto      M  43   80.0  183.0 ## vvkaj.00002 vvkaj.00002 VVKAJ101          2 Vegetarian      M  31   79.0  186.0 ## vvkaj.00023 vvkaj.00023 VVKAJ108          2       Keto      F  47   53.9  161.3 ##                  BMI Waist.Circumference ## vvkaj.00037 23.88844                98.0 ## vvkaj.00002 22.83501                80.0 ## vvkaj.00023 20.71667                85.5</code></td>
</tr>
<tr class="odd">
<td>Check the level 1 foods in your food tree. There should be nine categories.</td>
</tr>
<tr class="even">
<td><code>r L1s = tax_table(phyfoods)[, "L1"] as.vector(unique(L1s))</code></td>
</tr>
<tr class="odd">
<td><code>## [1] "L1_Milk_and_Milk_Products" ## [2] "L1_Meat_Poultry_Fish_and_Mixtures" ## [3] "L1_Eggs" ## [4] "L1_Dry_Beans_Peas_Other_Legumes_Nuts_and_Seeds" ## [5] "L1_Grain_Product" ## [6] "L1_Fruits" ## [7] "L1_Vegetables" ## [8] "L1_Fats_Oils_and_Salad_Dressings" ## [9] "L1_Sugars_Sweets_and_Beverages"</code></td>
</tr>
<tr class="even">
<td>### Use your phyloseq object and perform ordination</td>
</tr>
<tr class="odd">
<td>Change to the folder called “Unifrac” in your “VVKAJ” folder.</td>
</tr>
<tr class="even">
<td>Specify the directory where the data is.</td>
</tr>
<tr class="odd">
<td><code>r SpecifyDataDirectory(directory.name= "eg_data/VVKAJ/Unifrac/")</code> <!-- [NOTE] SpecifyDataDirectory function is not executed by R markdown, but just showing  --> <!--  the code so that this tutorial will have the same code as the R script.  --></td>
</tr>
<tr class="even">
<td>Perform Principal Coordinate Analysis (PCoA) with weighted unifrac distance of your food data. This may take a few minutes depending on your data size; e.g., a large phyloseq object (8 MB) takes ~ 1 min. Ordination by <em>UNweighted</em> unifrac distances can be done by <code>weighted=FALSE</code>.</td>
</tr>
<tr class="odd">
<td><code>r ordinated &lt;- phyloseq::ordinate(phyfoods, method="PCoA", distance="unifrac", weighted=TRUE)</code></td>
</tr>
<tr class="even">
<td>Save the percent variance explained by the axes as a vector to use in plots.</td>
</tr>
<tr class="odd">
<td><code>r eigen_percent &lt;- ordinated$values$Relative_eig</code></td>
</tr>
<tr class="even">
<td>Save the percent variance explained as a txt file.</td>
</tr>
<tr class="odd">
<td><code>r Eigen(eigen.input = eigen_percent, output.fn="4Lv_ordinated_Weighted_eigen_percent.txt")</code></td>
</tr>
<tr class="even">
<td>Merge the first n axes to the metadata and save it as a txt file. This will be used for plotting ordination results.</td>
</tr>
<tr class="odd">
<td><code>r MergeAxesAndMetadata(ord.object= ordinated, number.of.axes= 10, meta.data= meta, output.fn= "4Lv_ordinated_Weighted_meta_users.txt")</code></td>
</tr>
<tr class="even">
<td><br></td>
</tr>
</tbody>
</table>
</div>
</div>
<div id="save-unifrac-distance-unweighted-or-weighted-matrices." class="section level3">
<h3>Save unifrac distance (unweighted or weighted) matrices.</h3>
<p>Generate and save a weighted unifrac distance matrix of “Samples”.</p>
<pre class="r"><code>WeightedUnifracDis(input.phyloseq.obj= phyfoods, 
                   output.fn= &quot;4Lv_WEIGHTED_uni_dis.txt&quot;)        </code></pre>
<p>Generate and save an unweighted unifrac distance matrix of “Samples”.</p>
<pre class="r"><code>UnweightedUnifracDis(input.phyloseq.obj= phyfoods, 
                     output.fn= &quot;4Lv_UNweighted_uni_dis.txt&quot;)        </code></pre>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
